src/pages/error/NotFound.tsx

```diff
@@ -1,5 +1,40 @@
+import { Link, useNavigate } from 'react-router-dom';
+
+import VerticalLogo from '@/assets/images/VerticalLogo';
+import Button from '@/components/common/button';
+import UI_ERRORS from '@/constants/errors/uiErrors';
+import { ROUTES } from '@/constants/paths';
+
+/**
+ * @description 404 Not Found 에러 페이지.
+ * 존재하지 않는 URL 경로로 접근했을 때 표시되는 전용 페이지입니다.
+ * 다른 에러와 달리, 사용자에게 친근한 안내를 제공하는 데 초점을 맞춥니다.
+ */
 const NotFound = () => {
-  return <div>404 잘못된 경로를 입력했을때</div>;
+  const { TITLE, MESSAGE } = UI_ERRORS.NOT_FOUND;
+  const navigate = useNavigate();
+
+  return (
+    <div className='flex h-screen w-full flex-col items-center justify-center gap-y-16 bg-gray-100 text-center'>
+      <header>
+        <Link to={ROUTES.APP}>
+          <VerticalLogo aria-label='카바나 로고' className='h-auto w-64' />
+        </Link>
+      </header>
+
+      <main className='flex flex-col items-center gap-y-6'>
+        <h1 className='text-6xl font-extrabold text-capybara'>404</h1>
+        <div className='flex flex-col gap-y-2'>
+          <p className='text-2xl font-bold text-gray-800'>{TITLE}</p>
+          <p className='text-gray-600'>{MESSAGE}</p>
+        </div>
+      </main>
+
+      <footer className='mt-8'>
+        <Button onClick={() => navigate(ROUTES.APP, { replace: true })}>홈으로 돌아가기</Button>
```

Comment:
```suggestion
        <Button replace as={Link} to={ROUTES.APP}>홈으로 돌아가기</Button>
```
이 부분은 홈으로 이동하는 버튼이니 as={Link}를 사용해서 Link태그로 사용해도 되지 않을까요?

Reply:
여기는 replace: true로 히스토리를 초기화하기로 해서 링크대신 버튼을 써야했습니다.

Reply:
링크태그의 replace를 활용하는 쪽으로 수정했습니다.